File: atcoder.jp/contests/arc002/tasks/arc002_1/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

const isLeap = (y: number) =>
  y % 400 === 0 ? true : y % 100 === 0 ? false : y % 4 === 0;

console.log(isLeap(+input()) ? "YES" : "NO");

File: atcoder.jp/contests/arc002/tasks/arc002_1/main.rs
use proconio::input;

fn is_leap(y: i32) -> bool {
    match y {
        y if y % 400 == 0 => true,
        y if y % 100 == 0 => false,
        _ => y % 4 == 0,
    }
}

fn main() {
    input! {
        y: i32,
    }

    println!("{}", if is_leap(y) { "YES" } else { "NO" });
}
